mesonbuild/backend/ninjabackend.py

```diff
@@ -1597,6 +1600,64 @@ def generate_rust_target(self, target):
             self.generate_shsym(target)
         self.create_target_source_introspection(target, rustc, args, [main_rust_file], [])
 
+    def generate_zig_target(self, target):
+        zig = target.compilers['zig']
+        args = zig.compiler_args()
+
+        target_name = os.path.join(target.subdir, target.get_filename())
+
+        # Adapted from the rust target generation logic, Zig behaves similarly
+        # in that it only requires a single, so called "root source file"
```

Comment:
Will tweak, I originally wanted to use `build-obj` to make it possible to use a swap-in linker instead of the zig provided one but that's probably not something Meson will want to deal with.